receiver/googlecloudspannerreceiver/internal/statsreader/intervalstatsreader.go

```diff
@@ -0,0 +1,149 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statsreader
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/datasource"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/metadata"
+)
+
+const (
+	backFillIntervalDuration = time.Hour
+)
+
+type intervalStatsReader struct {
+	currentStatsReader
+	backFillEnabled   bool
+	lastPullTimestamp time.Time
+}
+
+func newIntervalStatsReaderWithMaxRowsLimit(
+	logger *zap.Logger,
+	database *datasource.Database,
+	metricsMetadata *metadata.MetricsMetadata,
+	backFillEnabled bool,
+	topMetricsQueryMaxRows int) *intervalStatsReader {
+
+	reader := currentStatsReader{
+		logger:                 logger,
+		database:               database,
+		metricsMetadata:        metricsMetadata,
+		topMetricsQueryMaxRows: topMetricsQueryMaxRows,
+		statement:              intervalStatsStatement,
+	}
+
+	return &intervalStatsReader{
+		currentStatsReader: reader,
+		backFillEnabled:    backFillEnabled,
+	}
+}
+
+func newIntervalStatsReader(
```

Comment:
I am not sure if you want to create 2 factory method for one parameter difference.

Reply:
What are other options here? Do you prefer to have one constructor and pass zero in case you do not have limitations by max rows.

Reply:
Reduced to one factory method.